gobblin-runtime/src/main/java/org/apache/gobblin/quality/DataQualityEvaluator.java

```diff
@@ -0,0 +1,221 @@
+                continue;
+            }
+
+            DataQualityStatus taskDataQuality = null;
+            String result = taskState.getProp(ConfigurationKeys.TASK_LEVEL_POLICY_RESULT_KEY);
+            taskDataQuality = DataQualityStatus.fromString(result);
+            if (taskDataQuality != DataQualityStatus.NOT_EVALUATED) {
+                log.debug("Data quality status of this task is: " + taskDataQuality);
+                if (DataQualityStatus.PASSED == taskDataQuality) {
+                    passedFilesCount++;
+                } else if (DataQualityStatus.FAILED == taskDataQuality){
+                    failedFilesCount++;
+                    jobDataQualityStatus = DataQualityStatus.FAILED;
+                } else {
+                    log.warn("Unexpected data quality status: " + taskDataQuality + " for task: " + taskState.getTaskId());
+                }
+            } else {
+                // Handle files without data quality evaluation
+                nonEvaluatedFilesCount++;
+                log.warn("No data quality evaluation for task: " + taskState.getTaskId());
+            }
+        }
+
+        // Log summary of evaluation
+        log.info("Data quality evaluation summary - Total: {}, Passed: {}, Failed: {}, Not Evaluated: {}",
+            totalFiles, passedFilesCount, failedFilesCount, nonEvaluatedFilesCount);
+        return new DataQualityEvaluationResult(jobDataQualityStatus, totalFiles, passedFilesCount, failedFilesCount, nonEvaluatedFilesCount);
+    }
+
+
+    private static void emitMetrics(JobState jobState, final DataQualityStatus jobDataQuality, final int totalFiles,
+            final int passedFilesCount, final int failedFilesCount, final int nonEvaluatedFilesCount, final String datasetUrn) {
+        try {
+            // Check if OpenTelemetry is enabled
+            boolean otelEnabled = jobState.getPropAsBoolean(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENABLED,
+                ConfigurationKeys.DEFAULT_METRICS_REPORTING_OPENTELEMETRY_ENABLED);
+
+            if (!otelEnabled) {
+                log.info("OpenTelemetry metrics disabled, skipping metrics emission");
+                return;
+            }
+
+            OpenTelemetryMetricsBase otelMetrics = OpenTelemetryMetrics.getInstance(jobState);
+            log.info("OpenTelemetry instance obtained: {}", otelMetrics != null);
+
+            if (otelMetrics != null) {
+                Attributes tags = getTagsForDataQualityMetrics(jobState, datasetUrn);
+                log.info("Tags for data quality metrics: " + tags.toString());
+                // Emit data quality status (1 for PASSED, 0 for FAILED)
+                log.info("Data quality status for this job is " + jobDataQuality);
+                if (jobDataQuality == DataQualityStatus.PASSED) {
+                    log.info("Data quality passed for job: {}", jobState.getJobName());
+                    otelMetrics.getMeter(GAAS_OBSERVABILITY_METRICS_GROUPNAME)
```

Comment:
we can avoid repeated getMeter(...) calls and assign the meter to a local variable once and reuse it for all metrics:
```
Meter meter = otelMetrics.getMeter(GAAS_OBSERVABILITY_METRICS_GROUPNAME);
```